app/app.go

```diff
@@ -169,34 +169,38 @@
 		var err error
 		p2pKey, err = p2p.LoadPrivKey(conf.DataDir)
 		if err != nil {
-			return nil, nil, errors.Wrap(err, "load p2p key")
+			return nil, nil, err
 		}
 	}
 
 	localEnode, peerDB, err := p2p.NewLocalEnode(conf.P2P, p2pKey)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "create local enode")
+		return nil, nil, err
 	}
 
-	udpNode, err := p2p.NewUDPNode(ctx, conf.P2P, localEnode, p2pKey, manifest.Peers)
+	bootnodes, err := p2p.NewUDPBootnodes(ctx, conf.P2P, manifest.Peers, localEnode.ID())
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "start discv5 listener")
+		return nil, nil, err
 	}
 
-	connGater, err := p2p.NewConnGater(manifest.PeerIDs(), udpNode.Relays)
+	udpNode, err := p2p.NewUDPNode(conf.P2P, localEnode, p2pKey, bootnodes)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "connection gater")
+		return nil, nil, err
 	}
 
-	tcpNode, err := p2p.NewTCPNode(conf.P2P, p2pKey, connGater, udpNode, manifest.Peers, p2p.EmptyAdvertisedAddrs)
+	relays, err := p2p.NewRelays(conf.P2P, bootnodes)
```

Comment:
this can return `nil,nil` why aren't we checking for nil for `relays`?

Reply:
`nil` is a valid empty slice, so not needed